File: src/main/java/com/dunamismax/site/config/SiteProperties.java
package com.dunamismax.site.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "site")
public record SiteProperties(
    String host,
    String title,
    String defaultDescription,
    String ogDescription) {

  public SiteProperties {
    if (host == null || host.isBlank()) {
      host = "dunamismax.com";
    }
    if (title == null || title.isBlank()) {
      title = "Stephen Sawyer · dunamismax";
    }
    if (defaultDescription == null || defaultDescription.isBlank()) {
      defaultDescription = "Stephen Sawyer is a Rust-first engineer and IT operator "
          + "focused on high-performance systems, crypto infrastructure, cryptography, "
          + "PostgreSQL, Python automation, privacy, and security.";
    }
    if (ogDescription == null || ogDescription.isBlank()) {
      ogDescription = "Rust-first systems, PostgreSQL-backed data, Python automation, "
          + "cryptography, encryption, self-hosting, privacy, and security.";
    }
  }
}

File: src/main/kotlin/com/dunamismax/site/config/SiteProperties.kt
package com.dunamismax.site.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "site")
data class SiteProperties(
    val host: String = "dunamismax.com",
    val title: String = "Stephen Sawyer · dunamismax",
    val defaultDescription: String =
        "Stephen Sawyer is an engineer working in Java, PostgreSQL, and " +
            "server-rendered web apps — open source advocate and privacy/security-minded " +
            "builder with 15 years in IT.",
    val ogDescription: String =
        "One language, one database, one VM. Java, PostgreSQL, " +
            "server-rendered HTML. Open source, self-hosting, privacy, and security.",
)
